sw/device/silicon_creator/manuf/skus/earlgrey_a0/sival_bringup/ft_personalize.c

```diff
@@ -244,6 +245,52 @@ static uint32_t get_cert_size(const uint8_t *header) {
   return (((uint32_t)header[2]) << 8) + header[3];
 }
 
+/**
+ *  Read a certificate from the passed in location in a flash INFO page and hash
+ *  its contents on the passed in SHA256 context. Determine the actual
+ *  certificate size from its ASN1 header.
+ *
+ * If the caller passed a pointer, save there the certificate size.
+ */
+static status_t hash_certificate(const flash_ctrl_info_page_t *page,
+                                 size_t offset, sha256_state_t *state,
+                                 size_t *size) {
+  uint8_t buffer[1024];  // 1K should be enough for the largest possible
+                         // certificate.
+  uint32_t cert_size;
+  uint32_t bytes_read;  // Number of bytes already read.
```

Comment:
is this variable need or can you just pass `offset + 1` to the second call to `flash_ctrl_info_read()`?

Reply:
Buffer pointer passed to flash_crtl_info_read() also needs to be updated, I think having this variable and a comment makes the code a bit better readable.

Reply:
ah sorry missed that.